pkg/service/service.go

```diff
@@ -241,15 +241,17 @@ func PushKubernetesResources(client kclient.ClientInterface, devfileObj parser.D
 
 // PushKubernetesResource pushes a Kubernetes resource (u) to the cluster using client
 // adding labels to the resource
-func PushKubernetesResource(client kclient.ClientInterface, u unstructured.Unstructured, labels map[string]string, annotations map[string]string) (bool, error) {
-	if isLinkResource(u.GetKind()) {
-		// it's a service binding related resource
-		return false, nil
+func PushKubernetesResource(client kclient.ClientInterface, u unstructured.Unstructured, labels map[string]string, annotations map[string]string, mode string) error {
+	sboSupported, err := client.IsServiceBindingSupported()
+	if err != nil {
+		return err
 	}
 
-	isOp, err := isOperatorBackedService(client, u)
-	if err != nil {
-		return false, err
+	// If the component is of Kind: ServiceBinding, trying to run in Dev mode and SBO is not installed, run it without operator.
+	if isLinkResource(u.GetKind()) && mode == odolabels.ComponentDevMode && !sboSupported {
```

Comment:
Why only for dev mode? Don't we want to use the library for `odo deploy`?

Reply:
With `odo deploy`, I would expect user to know that they're supposed to install ServiceBindingOperator if they intend to use ServiceBinding, just like they would with any other CRD.
Thinking about it this way, I think it makes sense to only check for `odo dev`, but if we want to provide consistency, having the same behavior for `odo deploy` makes sense.

Reply:
OK.

I can see some tests failing about odo logs, probably not related, I'll restart the tests to be sure

Reply:
No, they're failing locally as well. I am debugging them.